### PR TITLE
Charts for IRSA support with Helm

### PR DESCRIPTION
**Pull request by ryansteakley**
**Which issue is resolved by this Pull Request:**
Resolves #

**Description of your changes:**
Includes chart changes for IRSA for helm and basic implementation of installation and makefile changes to support installation. Putting it up for review, will continue to iterate on the installation code but need review on basics asap

**Testing:**
- [ ] Unit tests pass
- [ ] e2e tests pass
- Details about new tests (If this PR adds a new feature)
- Details about any manual tests performed

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jsitu777 — COMMENTED** [review]
Why this file is empty? (tests/e2e/utils/pipelines/__init__.py)

File: Makefile [hunk 1]
```diff
@@ -91,22 +91,27 @@ port-forward:
...
+
+bootstrap-pipelines: verify-cluster-variables connect-to-eks-cluster
+	yq e '.cluster.name=env(CLUSTER_NAME)' -i tests/e2e/utils/pipelines/config.yaml
+	yq e '.cluster.region=env(CLUSTER_REGION)' -i tests/e2e/utils/pipelines/config.yaml
+	cd tests/e2e && PYTHONPATH=.. python3 utils/pipelines/setup_pipelines_irsa.py
 
 cleanup-ack-req: verify-cluster-variables
 	yq e '.cluster.name=env(CLUSTER_NAME)' -i tests/e2e/utils/ack_sm_controller_bootstrap/config.yaml
 	yq e '.cluster.region=env(CLUSTER_REGION)' -i tests/e2e/utils/ack_sm_controller_bootstrap/config.yaml
-	cd tests/e2e && PYTHONPATH=.. python3.8 utils/ack_sm_controller_bootstrap/cleanup_sm_controller_req.py
```
Comment (jsitu777) [inline Makefile:99]: why change it from python 3.8 to python 3?

File: awsconfigs/apps/pipeline/s3-static/disable-minio-server-resources.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+$patch: delete
```
Comment (jsitu777) [inline awsconfigs/apps/pipeline/s3-static/disable-minio-server-resources.yaml:1]: This file shows up twice (one for s3-static and one for s3)

File: awsconfigs/apps/pipeline/s3-static/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  files:
+  - viewer-pod-template.json
+- name: kubeflow-pipelines-profile-controller-code
+  behavior: replace
+  files:
+  - sync.py
+generatorOptions:
+  disableNameSuffixHash: true
+patchesStrategicMerge:
+- disable-minio-server-resources.yaml
```
Comment (jsitu777) [inline awsconfigs/apps/pipeline/s3-static/kustomization.yaml:25]: consolidate this file so it only shows up once?

File: awsconfigs/apps/pipeline/s3-static/sync.py [hunk 4]
```diff
@@ -0,0 +1,423 @@
+# Copyright 2020-2021 The Kubeflow Authors
```
Comment (jsitu777) [inline awsconfigs/apps/pipeline/s3-static/sync.py:1]: is this script from the community? do you have a ref link?

**Review by jsitu777 — COMMENTED** [review]
Great work! left some comments.

File: tests/e2e/utils/pipelines/setup_pipelines_irsa.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            }
+        ],
+    }
+    return json.dumps(trust_policy)
+
+
+def create_pipeline_oidc_role(cluster_name, region):
+    iam_client = get_iam_client(region=region)
+    acc_id = get_account_id()
+    role_name = f"{common.PIPELINE_OIDC_ROLE_NAME_PREFIX}-{cluster_name}"
```
Comment (ananth102) [inline tests/e2e/utils/pipelines/setup_pipelines_irsa.py:58]: Can you limit the length of this string to 64 characters, IAM roles can have 64 [characters](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html#cfn-iam-role-rolename) but an eks cluster name can have upto 100 [characters](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-eks-cluster.html#cfn-eks-cluster-name)

File: tests/e2e/utils/pipelines/setup_pipelines_irsa.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    cfg = load_yaml_file(file_path=config_file_path)
+    cluster_region = cfg["cluster"]["region"]
+    cluster_name = cfg["cluster"]["name"]
+
+    print_banner("Create OIDC IAM role for Pipelines")
+    try:
+        create_pipeline_oidc_role(cluster_name, cluster_region)
+    except Exception as e:
+        print(e)
+        print("Try running cleanup_pipeline_irsa.py")
```
Comment (ananth102) [inline tests/e2e/utils/pipelines/setup_pipelines_irsa.py:101]: Does the code after get executed when there is an exception.

File: tests/e2e/utils/kubeflow_installation.py [hunk 7]
```diff
@@ -230,6 +250,28 @@ def install_ack_controller():
...
+        )
+
+    else:
+        IAM_ROLE_ARN_FOR_IRSA = cfg["pipeline_oidc_role"]
+        CHART_EXPORT_PATH = f"{installation_paths}/templates/ServiceAccount/ml-pipeline-kubeflow-ServiceAccount.yaml"
+        exec_shell(
+            f'yq e \'.metadata.annotations."eks.amazonaws.com/role-arn"="{IAM_ROLE_ARN_FOR_IRSA}"\' '
+            + f"-i {CHART_EXPORT_PATH}"
+        )
+        install_helm(component_name, installation_paths)
```
Comment (ananth102) [inline tests/e2e/utils/kubeflow_installation.py:272]: (Can do later) Would try to use the install_helm statement in the install_component function if possible.

Comment (ryansteakley) [reply tests/e2e/utils/kubeflow_installation.py:272]: changed to support this

File: awsconfigs/apps/pipeline/s3-static/sync.py (hunk 4 above)
Comment (ryansteakley) [reply awsconfigs/apps/pipeline/s3-static/sync.py]: mostly from community, the rest was added by meghna

File: Makefile (hunk 1 above)
Comment (ryansteakley) [reply Makefile:99]: will revert this got changed, in my user testing

File: awsconfigs/apps/pipeline/s3-static/kustomization.yaml (hunk 3 above)
Comment (ryansteakley) [reply awsconfigs/apps/pipeline/s3-static/kustomization.yaml]: done

File: tests/e2e/utils/pipelines/setup_pipelines_irsa.py (hunk 6 above)
Comment (ryansteakley) [reply tests/e2e/utils/pipelines/setup_pipelines_irsa.py]: yes it does, this is mostly in the situation where the oidc_role already existed so it failed to create it again, but we may still need to output the information

File: tests/e2e/utils/pipelines/cleanup_pipeline_irsa.py [hunk 8]
```diff
@@ -1,54 +0,0 @@
...
-    cfg = load_yaml_file(file_path=config_file_path)
-    cluster_region = cfg["cluster"]["region"]
-    cluster_name = cfg["cluster"]["name"]
-
-    print_banner("Deleting all resources created for Pipeline IRSA")
-    role_name = f"{common.PIPELINE_OIDC_ROLE_NAME_PREFIX}-{cluster_name}"
-    delete_iam_role(role_name, cluster_region)
-
-    print_banner("CLEANUP SUCCESSFUL")
-
```
Comment (jsitu777) [inline tests/e2e/utils/pipelines/cleanup_pipeline_irsa.py:54]: the clean up script is deleted on purpose?

Comment (ryansteakley) [reply tests/e2e/utils/pipelines/cleanup_pipeline_irsa.py:54]: yes, pushed two of the same script named differnently.

**Review by jsitu777 — APPROVED** [review]

**Comment by ananth102** [comment]
looks good, can you also trim the role name here https://github.com/awslabs/kubeflow-manifests/blob/efd26dac882b7865ee2e50641beabd8c6f480a7c/tests/e2e/utils/pipelines/setup_pipelines_irsa.py#L58

**Review by ananth102 — APPROVED** [review]

**Pull request merged**